code/tier_classification/tier_classification.R

```diff
@@ -145,29 +144,69 @@ if(nrow(oncokb_anno) == 0){
+    
+    key_findings_output_snv <- key_clinical_findings_output %>% 
+      dplyr::filter(grepl("Missense_Mutation", Details) | grepl("Nonsense_Mutation", Details) | grepl("Splice_Site", Details) | grepl("Splice_Region", Details)) %>%
+      dplyr::mutate(position = gsub(".*HGVSp: p..", "", Details)) %>% 
+      dplyr::mutate(position =str_sub(position, end = -2))
+      
     all_findings_output_hotspots <- data.frame()
     key_clinical_findings_output_hotspots <- data.frame()
     
-    for(p in 1:nrow(hotspot_snv)){
+    snv_location <- hotspot_snv %>% 
+      dplyr::select(Hugo_Symbol, Amino_Acid_Position) %>% 
+      distinct()
+    
+    for(p in 1:nrow(snv_location)){
       
-      each_entry <- hotspot_snv[p,]
+      each_entry <- snv_location[p,]
       hotspot_gene <- each_entry$Hugo_Symbol
       hotspot_aa_position <- each_entry$Amino_Acid_Position
       
-      all_findings_output_hotspot <- all_findings_output %>%
-        dplyr::filter(Aberration == hotspot_gene & grepl(hotspot_aa_position, Details)) %>%
-        dplyr::filter(grepl("Missense_Mutation", Details) | grepl("Nonsense_Mutation", Details) | grepl("Splice_Site", Details) | grepl("Splice_Region", Details))%>%
-        dplyr::mutate(Variant_Properties= paste0("Cancer Hotspot Location; ", Variant_Properties))
+      # find all the specific AA mutations in the hotspot table
+      hotspot_aa <- hotspot_snv %>% 
+        dplyr::filter(Amino_Acid_Position == hotspot_aa_position,
+                      Hugo_Symbol==hotspot_gene) %>%
+        dplyr::pull(HGVSp_short)
       
-      all_findings_output_hotspots <- bind_rows(all_findings_output_hotspots, all_findings_output_hotspot)
+      # reformat to make it matchable 
+      hotspot_aa<-paste(noquote(hotspot_aa),collapse='|')
       
-      key_clinical_findings_output_hotspot <- key_clinical_findings_output %>% 
-        dplyr::filter(Aberration == hotspot_gene & grepl(hotspot_aa_position, Details)) %>%
-        dplyr::filter(grepl("Missense_Mutation", Details) | grepl("Nonsense_Mutation", Details) | grepl("Splice_Site", Details) | grepl("Splice_Region", Details))%>%
-        dplyr::mutate(Variant_Properties= paste0("Cancer Hotspot Location; ", Variant_Properties))
+      # first deal with all clinical findings table
+      all_findings_output_hotspot <- all_findings_output_snv %>%
+        dplyr::filter(Aberration == hotspot_gene,
+                      position == hotspot_aa_position) %>%
+        dplyr::select(-position)
       
-      key_clinical_findings_output_hotspots <- bind_rows(key_clinical_findings_output_hotspots, key_clinical_findings_output_hotspot)
+      if(nrow(all_findings_output_hotspot)>0){
+        all_findings_output_hotspot <- all_findings_output_hotspot %>%
+          dplyr::mutate(Variant_Properties= case_when(
+            grepl(hotspot_aa, Details) ~ paste0("Cancer Hotspot; ", Variant_Properties),
+            TRUE ~ paste0("Cancer Hotspot Location; ", Variant_Properties)
```

Comment:
I'm not 100% clear on this line - will this still annotate variants with 'cancer hotspot location' if the chromosomal location of the variant matches but not the actual AA change?